WordPress/Classes/Extensions/UIImageView+SiteIcon.swift

```diff
@@ -0,0 +1,159 @@
+        }
+    }
+
+
+    /// Downloads the SiteIcon Image, hosted at the specified path. This method will attempt to optimize the URL, so that
+    /// the download Image Size matches `SiteIconDefaults.imageSize`.
+    ///
+    /// TODO: This is a convenience method. Nuke me once we're all swifted.
+    ///
+    /// - Parameter path: SiteIcon's url (string encoded) to be downloaded.
+    ///
+    @objc
+    func downloadSiteIcon(at path: String) {
+        downloadSiteIcon(at: path, placeholderImage: .siteIconPlaceholderImage)
+    }
+
+
+    /// Downloads the SiteIcon Image, hosted at the specified path. This method will attempt to optimize the URL, so that
+    /// the download Image Size matches `SiteIconDefaults.imageSize`.
+    ///
+    /// - Parameters:
+    ///     - path: SiteIcon's url (string encoded) to be downloaded.
+    ///     - placeholderImage: Yes. It's the "place holder image", Sherlock.
+    ///
+    @objc
+    func downloadSiteIcon(at path: String, placeholderImage: UIImage?) {
+        guard let siteIconURL = optimizedURL(for: path) else {
+            image = placeholderImage
+            return
+        }
+
+        setImageWith(siteIconURL, placeholderImage: placeholderImage)
+    }
+
+
+    /// Downloads the SiteIcon Image, associated to a given Blog. This method will attempt to optimize the URL, so that
+    /// the download Image Size matches `SiteIconDefaults.imageSize`.
+    ///
+    /// - Parameters:
+    ///     - blog: reference to the source blog
+    ///     - placeholderImage: Yes. It's the "place holder image".
+    ///
+    @objc
+    func downloadSiteIcon(for blog: Blog, placeholderImage: UIImage? = .siteIconPlaceholderImage) {
+        guard let siteIconPath = blog.icon, let siteIconURL = optimizedURL(for: siteIconPath) else {
+            image = placeholderImage
+            return
+        }
+
+        if !blog.isPrivate() {
+            setImageWith(siteIconURL, placeholderImage: placeholderImage)
+            return
+        }
+
+        let request = PrivateSiteURLProtocol.requestForPrivateSite(from: siteIconURL)
+        setImageWith(request, placeholderImage: placeholderImage, success: nil, failure: nil)
+    }
+}
+
+
+// MARK: - Private Methods
+//
+private extension UIImageView {
+
+    /// Returns the Size Optimized URL for a given Path.
+    ///
+    func optimizedURL(for path: String) -> URL? {
+        if isPhotonURL(path) || isDotcomURL(path) {
+            return optimizedDotcomURL(from: path)
+        }
+
+        if isBlavatarURL(path) {
+            return optimizedBlavatarURL(from: path)
+        }
+
+        return optimizedPhotonURL(from: path)
+    }
+
+
+    // MARK: - Private Helpers
+
+    /// Returns the download URL for a square icon with a size of `SiteIconDefaults.imageSizeInPixels`
+    ///
+    /// - Parameter path: SiteIcon URL (string encoded).
+    ///
+    private func optimizedDotcomURL(from path: String) -> URL? {
```

Comment:
The extension is already private, so you could save some extra `private` keywords from these functions 😉

Reply:
This is actually on purpose, and removing that `private` keyword would produce a different result. The pattern is as follows:

```
class A {
    func doSomething() {
        doSomethingInternal()
    }
}

private extension A {
   func doSomethingInternal() {
       doSomethingPrivate()
   }

   private func doSomethingPrivate() { }
}
```

**doSomethingInternal** is accessible anywhere within the file. Now: **doSomethingPrivate** is only accessible within *the extension in which it's declared*.

If **doSomething** was to call **doSomethingPrivate()**, the compiler would complain. In our case (UIImageView+SiteIcon), this "double private" is actually on purpose, to keep those methods private to the extension in which they're declared.

Reply:
I see what you mean, but those methods would still be fine as `fileprivate`. I don't think it's really important though 🤷‍♂️ 😁

Reply:
Sooort of. Changing the visibility would expose them to other snippets within the same file. We used this technique in Aztec, to encapsulate ... (facepalm) UIKit workarounds, within the same file, while keeping a single entry point.